src/languages/ini.js

```diff
@@ -66,7 +66,7 @@ export default function(hljs) {
         begin: /\[+/, end: /\]+/
       },
       {
-        begin: /^[a-z0-9\[\]_\.-]+(?=\s*=\s*)/,
+        begin: /([A-Za-z0-9_-]+|(\s*\.\s*)|"(\\"|[^"])*"|'[^']*')+(?=\s*=\s*)/,
```

Comment:
Can we please use `variants` for this to make it more obvious what is happening here, what the different flavors of tags are supported, etc.

Reply:
Is A-Z correct? I don't see any uppercase keys in the test?

Reply:
Can you please explain `(\s*\.\s*)`?

Reply:
> Can you please explain `(\s*\.\s*)`?

According to the TOML spec, "whitespace around dot-separated parts is ignored".

> Is A-Z correct? I don't see any uppercase keys in the test?

It is in TOML, let me add a test for that.

Reply:
You removed the `^` so I'm worried about this matching who knows what now... Why is the ^ not correct?

Reply:
The `^` was preventing indented keys to be selected:

```toml
key = "value"
  indented_key = "value"
```

I originally thought I could add a `\s*` to fix that, but the whitespace is not actually part of the key so I'm not sure. What do you think?

Reply:
What are the full rules for values... what would this be/do?

```tom
indented_key = amiakey = "value"
```

Reply:
> what would this be/do?
> 
> ```
> indented_key = amiakey = "value"
> ```

That's not valid TOML, though without the lookbehind, `amiakey` would be highlighted as a key..